5269 TOGGLE DAPI Journalfore alle saker

janitamarielle opened the pull request:
Har prøvd å være beskrivende med commit-meldingene, men for å **oppsummere endringene**: 
- Opprett ny sak: 
    - Nye felt Sakstema og Behandlingstype. Disse sendes også til backend.
    - Sakstema/behandlingstema er ikke lengre pre-utfylt.
    - Gjemmer periode/land når sakstema ikke er EØS eller når vi journalfører på virksomhet.
- Knytt til eksisterende sak:
    - Nytt design på eksisterende saker. 
    - Begrensning på viste saker, og mulighet for å vise alle
    - Fjerner tekst "Knytt til sak"
    - Gjør komponent for valg av behandlingstype om fra Select til Radio
- Radio-buttons for å skille mellom "Eksisterende saker" og "Opprett ny sak" istedenfor at Opprett ny sak kom nederst på eksisterende-saker-listen.
- Nytt design når det ikke finnes eksisterende saker på bruker/virksomhet.

**Alle endringene skal være togglet**. Eneste unntaket er små css-endringer.

Merk at skissen og jira ikke helt stemmer. Ta gjerne kontakt dersom det er usikkerhet - måtte selv høre med fag. Jira blir oppdatert på mandag. 

Forventer noen endringer i objekter fra melosys-api, men er togglet så kan godt prodsettes uten: https://github.com/navikt/melosys-api/pull/1320

Jira: https://jira.adeo.no/browse/MELOSYS-5269

Skisser: https://www.figma.com/file/xYpUVr1mz7gQKIzULO46I4/Journalf%C3%B8ringer?node-id=1138%3A27807

janitamarielle committed:
Legger til feltene sakstema og behandlingstype når toggle melosys.sakstema er enabled. Da er disse feltene også required i skjema. Grunnet sistnevnte her valgte jeg å sende nedover visSakstema - slik at skjema fikk tilgang til verdien.
Sender med disse verdiene til backend.

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4798_journalfore-alle-saker

janitamarielle committed:
Knytt til eksisterende sak - alternativt design

janitamarielle committed:
Endret fra visSakstema til sakstemaToggleEnabled for å klargjøre variabel.
Dersom sakstema er toggled på skal man kun vise "Journalfør uten videre behandling" for EUEØS flyten.

janitamarielle committed:
Legger på funksjonalitet for å begrens viste saker av gangen. Gjemmer denne bak toggle.
journalforing.js venter på toggle er hentet før den viser noe, denne kan fjernes når toggle fjernes da den hjelper til med begrensVisning-funksjonaliteten mens den er togglet.

janitamarielle committed:
La til nye radioknapper for å skille mellom knytte til eksisterende saker og opprett ny sak. Legger også denne bak toggle

janitamarielle committed:
Annen visning for ingen saker etter sakstema er togglet på.
Ble mye å vente med å laste hele journalforingsbildet til toggle ble hentet. Flyttet den litt nedover.

janitamarielle committed:
Sakstype og behandlingstema skal ikke være utfylt etter toggle blir enabled.

janitamarielle committed:
Småtteri for styling av opprettsak

janitamarielle committed:
Knytt til sak komponent endres i henhold til beskrivelse av fag. Fjerne tekst "Knytt til sak", gjøre select om til radios. La også på margin-left i henhold til skisser.
sakstype og opprettsak_behandlingstype er ikke required når vi skal knytte sak eller opprette ny vurdering.
Både eksisterende saker og opprett ny sak har behandlingstype, endrer opprettelse av dto mot backend for å unngå forvirring/mulig feil.

janitamarielle committed:
Land og periode skal ikke vises for virksomhet. Skrev også om til å sjekke sakstype istedenfor behandlingstema for FTLR og Trygdeavtale

janitamarielle committed:
Kan trykke på større deler av knappen

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4798_journalfore-alle-saker

janitamarielle committed:
Bruker den nye BehandlingsstatusMedSvarfrist, og viser fullt navn for landene

fredrik-oh commented on src/sider/journalforing/komponenter/opprettSak.js:
```diff
@@ -39,13 +40,13 @@ export const OpprettSakTittel = () => (
 );
 
 const OpprettFagsak = (props) => {
```
Rename filen til OpprettFagsak eller const til OpprettSak :)

janitamarielle committed:
Lik navn på fil og komponent: OpprettSak

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4798_journalfore-alle-saker

janitamarielle committed:
Skal velge behandlingstema også. Denne settes automatisk til den forrige behandlings behandlingstema.

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4798_journalfore-alle-saker

 Conflicts:
	src/sider/journalforing/komponenter/fagsakVelger.js
	src/sider/journalforing/komponenter/knyttTilSak.js

janitamarielle committed:
Flytter valgbareBehandlinger inn i knyttTilSak.js siden det er der det hører til

janitamarielle committed:
Bytter rekkefølgen på tema og type, og setter logikk på mulige behandlingstema og behandlingstyper.

janitamarielle committed:
Logikk for saker med status henlagt eller henlagt_bortfalt

janitamarielle committed:
Sender med behandlingstema for alle hensikter.

janitamarielle committed:
Fikser tester

janitamarielle committed:
Skriver test for henlagt visning

janitamarielle committed:
Må resette behandlingstype også siden denne styrer hensikten ved knytting

fredrik-oh commented on src/sider/journalforing/komponenter/knyttTilSak.js:
```diff
@@ -1,60 +1,152 @@
 
-  const clsElementskrift = { "border-bottom": "none" };
+  useEffect(() => {
+    return () => {
```
Kan vel flytte cleanup inn i den andre useEffecten?

janitamarielle replied:
Det blir ikke vanskelig å se at det er "onUnmount" som kaller cleanup da?

fredrik-oh replied:
Tja, så lenge man er kjent med hva return i en effect gjør blir  vel ikke koden mindre lesbar? Ikke en stor sak uansett så du bestemmer :)

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4798_journalfore-alle-saker

janitamarielle committed:
Saksnummer må settes ved bytting av valgtVisning, da saksnummer brukes til disable-logikk på journalføringsknapp.
Tok meg også friheten å bruke konstanten til form-navnet istedenfor direkte.

janitamarielle commented:
Test gjerne at det også ser bra ut med toggle av 😄

janitamarielle committed:
manglet en else

fredrik-oh commented on src/sider/journalforing/komponenter/knyttTilSak.js:
```diff
@@ -1,60 +1,152 @@
+    sisteBehandling.behandlingstema.kode
+  ) && sisteBehandling.behandlingstype.kode === MKVBehandlingstyper.FØRSTEGANG;
+
+const valgbareBehandlingstyper = (sakstype, behandlingstema, sisteBehandling) =>
```
Litt uklart for meg (utifra navnene) forskjellen på behandlingstyper og valgbareBehandlingstyper

janitamarielle replied:
Er bare gammel og ny metodenavn. Den gamle skal fjernes ved toggle-cleanup så jeg så ikke vitsen i å tenke mer på navngivning 🤷

janitamarielle committed:
Gjør marginen mellom legend på customRadioPanelGruppe.js slik at den kun gjelder fagsakVelger og ikke overalt i koden.
Merket at behandlingstyper forventet kode og fikk objekt. Gjorde slik at funksjonene forventer kode alltid.
